source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,620 @@
+typedef uint8_t tx_buffer_t[SDK_SIZEALIGN(ENET_TXBUFF_SIZE, FSL_ENET_BUFF_ALIGNMENT)];
+
+/*!
+ * @brief Used to wrap received data in a pbuf to be passed into lwIP
+ *        without copying.
+ * Once last reference is released, buffer can be used by ENET RX DMA again.
+ */
+typedef struct rx_pbuf_wrapper
+{
+    void *buffer;              /*!< Original buffer wrapped. */
+    volatile bool buffer_used; /*!< Wrapped buffer is used by ENET or FreeRTOS+TCP. */
+} rx_pbuf_wrapper_t;
+
+/**
+ * Helper structure to hold private data used to operate your Ethernet interface.
+ */
+struct ethernetif
+{
+    ENET_Type *base;
+    enet_handle_t handle;
+    enet_rx_bd_struct_t *RxBuffDescrip;
+    enet_tx_bd_struct_t *TxBuffDescrip;
+    rx_buffer_t *RxDataBuff;
+    tx_buffer_t *TxDataBuff;
+    rx_pbuf_wrapper_t RxPbufs[ENET_RXBUFF_NUM];
+};
+
+
+static mdio_handle_t mdioHandle = { .ops = &EXAMPLE_MDIO_OPS };
+
+static phy_handle_t phyHandle = { .phyAddr = 0x00, .mdioHandle = &mdioHandle, .ops = &EXAMPLE_PHY_OPS };
+
+/*-----------------------------------------------------------*/
+
+/*
+ * A deferred interrupt handler task that processes
+ */
+
+static TaskHandle_t receiveTaskHandle;
+
+static struct ethernetif EthernetInterface1;
+struct ethernetif * ethernetifLocal = &EthernetInterface1;
+
+/*-----------------------------------------------------------*/
+
+/* The queue used to communicate Ethernet events with the IP task. */
+extern QueueHandle_t xNetworkEventQueue;
+
+/*-----------------------------------------------------------*/
+
+AT_NONCACHEABLE_SECTION_ALIGN( static enet_rx_bd_struct_t rxBuffDescrip_0[ ENET_RXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+AT_NONCACHEABLE_SECTION_ALIGN( static enet_tx_bd_struct_t txBuffDescrip_0[ ENET_TXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+SDK_ALIGN( static rx_buffer_t rxDataBuff_0[ ENET_RXBUFF_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+SDK_ALIGN( static tx_buffer_t txDataBuff_0[ ENET_TXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+
+static void prvEMACHandlerTask( void * pvParameters );
+
+static void ethernet_callback( ENET_Type *base,
+                               enet_handle_t *handle,
+                               enet_event_t event,
+                               enet_frame_info_t *frameInfo,
+                               void *userData );
+
+static void prvProcessFrame( int length );
+
+BaseType_t xNetworkInterfaceInitialise( void )
+{
+	status_t xStatus;
+	phy_config_t xConfig = { 0 };
+	BaseType_t xResult = pdPASS;
+	phy_speed_t speed;
+    phy_duplex_t duplex;
+    status_t readStatus;
+    enet_config_t config;
+    uint32_t sysClock;
+    enet_buffer_config_t buffCfg[ENET_RING_NUM];
+    int i;
+    uint8_t ucCounter = 0;
+    bool autoNegotiationComplete = false;
+    bool LinkUp = false;
+
+    configASSERT( FSL_FEATURE_ENET_QUEUE == 1 );
+
+    #if defined( ENET_ENHANCEDBUFFERDESCRIPTOR_MODE )
+        configASSERT( pdFALSE );
+	#endif
+
+    ethernetifLocal->RxBuffDescrip = &( rxBuffDescrip_0[ 0 ] );
```

Comment:
Mind you that `xNetworkInterfaceInitialise()` may get called many times, while this initialisation only needs to be done a single time. When it fails, `xNetworkInterfaceInitialise()` should return `pdFAIL` ever after.
I like [this example](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/source/portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c#L197) in the driver for Ultrascale, it passes through these stages:
~~~
typedef enum xEMAC_STATE
{
    xEMAC_Init,
    xEMAC_SetupPHY,
    xEMAC_WaitPHY,
    xEMAC_Ready,
    xEMAC_Fatal,
} EMACState_t;
~~~
After `xEMAC_Init`, the new stage will either be `xEMAC_SetupPHY` or `xEMAC_Fatal`.

Reply:
Yes, I switched to using that.